kubernetes/nodecollection.go

```diff
@@ -415,17 +415,28 @@ func ensureNodeSource(ctx context.Context, config KubeAgentConfig) (KubeAgentCon
 		return config, FatalNodeError
 	}
 
+	validateConfig(config, proxyNodes, directNodes)
 	return config, nil
 }
 
+func validateConfig(config KubeAgentConfig, proxyNodes, directNodes int32) {
+	if proxyNodes > 0 {
+		config.NodeMetrics.SetAvailability(NodeStatsSummaryEndpoint, Proxy, true)
+	} else if directNodes > 0 {
```

Comment:
Should this be an `if` instead of an `else if`? Looks like SetAvailability can handle multiple connection methods

Reply:
It can take multiple but the downstream logic in the agent doesn't really support both being enabled. In `retrieveNodeData` if both were to be set, but a node only supported proxy the direct request would be attempted first, fail, and then return an error.